lib/active_admin/views/components/scopes.rb

```diff
@@ -15,11 +16,13 @@ def tag_name
       def build(scopes, options = {})
         super({ role: "toolbar" })
         add_class "scopes"
+        prepare_async_counts(scopes, options)
+
         scopes.group_by(&:group).each do |group, group_scopes|
           div class: "index-button-group", role: "group", data: { "group": group_name(group) } do
-            group_scopes.each do |scope|
-              build_scope(scope, options) if call_method_or_exec_proc(scope.display_if_block)
-            end
+            group_scopes
+              .select { |scope| display_scope?(scope) }
+              .each { |scope| build_scope(scope, options) }
```

Comment:
I would keep a single iteration. Performance is not the main argument because I belive scopes is a small collection. But by introducing a select before the each iteration it's not so evident that the main change the introduction of the new helper `display_scope?`

```suggestion
            group_scopes.each do |scope|
              build_scope(scope, options) if display_scope?(scope)
            end
```

Reply:
Done in [7dd372e](https://github.com/activeadmin/activeadmin/pull/8394/commits/7dd372e8beffdf498d181375459dc7689a67e093)